src/app/contribute/guidelines/components/index.html

```diff
@@ -0,0 +1,201 @@
+  </p> 
+  
+  <p>
+    If you do not document your components, the SKY UX team will create the documentation, but we request that you <a href="https://github.com/blackbaud/skyux2-docs/issues">file an issue</a>. However, we strongly encourage you to create your own documentation because it can be useful to uncover issues with your components.
+  </p>
+
+  <stache-page-anchor>
+    Write unit tests
+  </stache-page-anchor>
+
+  <p>
+    Your unit test files should end in <stache-code>.spec.ts</stache-code> and reside in the same folders as the components that they test. For example, the test file for <stache-code>foo/foo.component.ts</stache-code> should be <stache-code>foo/foo.component.spec.ts</stache-code>. To run unit tests for a library, run <stache-code>skyux test --coverage library</stache-code>. All test fixtures such as sample components, mock services, etc. should be in a fixtures folder within the component's folder. The existing codebase contains many examples of this pattern, so just follow the patterns in the existing code when you write tests.
+  </p>
+
+  <p>
+    For new components, we recommend that you reach out to the SKY UX team for a "soft" review of the initial implementation before you write unit tests. This allows you to make requested changes to your code before writing the tests.
+  </p>
+
+  <p>
+    When your tests run, code coverage results are generated and can be located in <stache-code>coverage/&lt;browser version>/index.html</stache-code>. You can launch this straight from disk and view the SKY UX unit test code coverage results in your default web browser.
+  </p>
+
+  <stache-page-anchor>
+    Write visual regression tests
+  </stache-page-anchor>
+
+  <p>
+    During continuous integration builds, SKY UX runs visual regression tests through BrowserStack using <a href="http://www.protractortest.org/">Protractor</a> and <a href="https://github.com/koola/pix-diff">pix-diff</a>. These prerequisites are automatically installed when you run <stache-code>skyux e2e</stache-code> to create and compare screenshots in the <stache-code>skyux-spa-visual-tests/screenshots-baseline-local/</stache-code> folder.
+  </p>
+
+  <p>
+    To create visual tests for a new component, create a folder for the tests in <stache-code>./src/app/visual</stache-code> and add four files for the visual regression test to the root <stache-code>e2e</stache-code> folder:
```

Comment:
I see another outdated but unresolved comment from Bobby about updating this in light of the separate modules we have now. Did anyone ever follow up on this? Bobby pinged @Blackbaud-SteveBrush, @Blackbaud-AlexKingman , and @Blackbaud-TrevorBurch in his comment, but I don't see any responses.

Reply:
Here's an updated version for the new structure. Let me know if you have any questions!

To create visual tests for a new component, follow these steps:

1. Create a folder in `./src/app/visual` that contains a visual demonstration of your component. It should contain the following files:
    1. `{componentName}-visual.component.html`: The template for the visual component to take a screenshot of.
    2. `{componentName}-visual.component.ts`: The TypeScript code for the visual component to take a screenshot of.
    3. `index.html`: The route file containing your new component.
2. Add a convenience link to this new folder to the `./src/app/index.html` file.
3. Add your new tests to the e2e suite by creating a file at `./e2e/{componentName}.e2e-spec.ts`